packages/react-native/src/InAppMessaging/components/BannerMessage/__tests__/BannerMessage.spec.tsx

```diff
@@ -83,39 +81,60 @@ describe('BannerMessage', () => {
         expectedProps: { children: 'body content' },
       },
     ],
+  ])(
+    'correctly handles a %s prop',
+    (key, testID, { testProps, expectedProps }) => {
+      mockUseMessageImage.mockReturnValueOnce({
+        hasRenderableImage: false,
+        imageDimensions: { height: undefined, width: undefined },
+        isImageFetching: false,
+      });
+
+      const props = { ...baseProps, [key]: testProps };
+
+      const { getByTestId } = render(<BannerMessage {...props} />);
+      const testElement = getByTestId(testID);
+
+      for (const [key, value] of Object.entries(expectedProps)) {
+        expect(testElement.props[key]).toEqual(value);
+      }
```

Comment:
Prefer explicit assertions in unit tests over iterative assertions which require the reader to spend more time understanding the unit test. Looks like the only thing we are asserting against here is `children` in `expectedProps` so should be fairly straightforward to make the assertion explicit?